src/compiler/parser.ts

```diff
@@ -6131,6 +6131,14 @@ namespace ts {
                 return jsDocTypeExpression ? { jsDocTypeExpression, diagnostics } : undefined;
             }
 
+            function parseUnbracketedJSDocTypeExpression(): JSDocTypeExpression {

```

Comment:
This could be done in `parseJSDocTypeExpression` like we do with `requireBraces`, which happens to be no longer used!